# Tarea 7
## Normality tests

In [1]:
# Read the file into a DataFrame: df
def import_csv(file):
    # Read the file into a DataFrame: df
    import pandas as pd
    return pd.read_csv(file)

In [2]:
# Path to the file to be imported
path = "../data/Terminos_lagoon_TA_DIC_2023_RawData.csv"

# Import the file
terminos = import_csv(path)

print(terminos.head())

   sample      date     estuary   area station layer_depth season  \
0  CDL01S  5/3/2020  Candelaria  River   CDL01     Surface    Dry   
1  CDL01F  5/3/2020  Candelaria  River   CDL01      Bottom    Dry   
2  CDL02S  5/3/2020  Candelaria  River   CDL02     Surface    Dry   
3  CDL02F  5/3/2020  Candelaria  River   CDL02      Bottom    Dry   
4  CDL03S  5/3/2020  Candelaria  River   CDL03     Surface    Dry   

   chlorophy_microg_l  cond_microsiemens_cm  depth_m  ...  do_mg_l  sal_psu  \
0                0.36                7015.4    0.464  ...     7.12     3.56   
1                4.19               29886.1    7.792  ...     4.90    16.97   
2                0.92               16691.1    0.453  ...     6.99     8.94   
3                2.23               24847.4    1.261  ...     6.52    13.87   
4                0.58               46341.6    0.465  ...     6.24    28.06   

   sp_cond_microsiemens_cm  turbidity_fnu  temp_c  latitude  longitude  \
0                   6547.7          

### Shapiro Wilks test

In [3]:
from scipy.stats import shapiro

ta = terminos['ta_micromol_kg']

stat, p = shapiro(ta)
print(f'Estadístico= {stat:.5f}, p-valor= {p:.5f}')
if p > 0.05:
    print("La distribución es normal (no se rechaza H0)")
else:
    print("La distribución NO es normal (se rechaza H0)")

Estadístico= 0.87971, p-valor= 0.00000
La distribución NO es normal (se rechaza H0)


In [4]:
#ver el valor de p
print(p)

9.071435657097518e-08


### Exercise
When is it appropriate to use the Kolmogorov-Smirnov test instead of the Shapiro-Wilk test?

the Kolmogorov-Smirnov test is appropiate to use when you need to compare two distributions and when the sample size is large, besides its needed to known already population distribution parameters such as mean and sd and not the estimated ones from sample data.

In [5]:
print(terminos.shape)

(106, 21)


only 106 data for each variable it´s debatable if the n size is enough, but we dont know distribution values of origin population or reference data, so we apply a shapiro test for each variable (DIC and salinity)

DIC shapiro test

In [8]:
from scipy.stats import shapiro

dic = terminos['dic_micromol_kg']

stat, p2 = shapiro(dic)
print(f'Estadístico= {stat:.5f}, p-valor= {p:.5f}')
if p2 > 0.05:
    print("La distribución es normal (no se rechaza H0)")
else:
    print("La distribución NO es normal (se rechaza H0)")

Estadístico= 0.85997, p-valor= 0.00000
La distribución NO es normal (se rechaza H0)


In [9]:
print(p2)

1.3836938477781853e-08


Salinity shapiro test

In [10]:
from scipy.stats import shapiro

sal = terminos['sal_psu']

stat, p3 = shapiro(dic)
print(f'Estadístico= {stat:.5f}, p-valor= {p:.5f}')
if p3 > 0.05:
    print("La distribución es normal (no se rechaza H0)")
else:
    print("La distribución NO es normal (se rechaza H0)")

Estadístico= 0.85997, p-valor= 0.00000
La distribución NO es normal (se rechaza H0)


In [11]:
print(p3)

1.3836938477781853e-08
